src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageRestoreBar.xaml.cs

```diff
@@ -215,10 +218,14 @@ private void OnRestoreLinkClick(object sender, RoutedEventArgs e)
             {
                 if (_projectContextInfo?.ProjectStyle == ProjectModel.ProjectStyle.PackageReference && _isAssetsFileMissing)
                 {
+                    var evt = new RestoreBannerClickedTelemetryEvent(RestoreButtonAction.MissingAssetsFile, GetRestoreButtonOriginTelemetry());
```

Comment:
A different way of doing this would be to add more details in the `vs/nuget/restoreinformation` event. 

Currently it has a RestoreOperationSource, https://github.com/NuGet/NuGet.Client/blob/e70d3c6167fc1c878acee98050784726152e4a31/src/NuGet.Clients/NuGet.VisualStudio.Common/Telemetry/RestoreTelemetryEvent.cs#L93. 

This would normally fall under `Explicit`, so you can add an extra property that can give  you more details. 

Advantages of this are that 
- you'd be able to reach the actual restore events.
- you'd natively only get PackageReference events as that's what the feature is about. 
- It's a lot easier if you don't have to do event joins.

Reply:
Note the reason why I wouldn't recommend OperationSource changes is that we already have the same pattern with implcit restore caused https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Clients/NuGet.VisualStudio.Common/Telemetry/ImplicitRestoreReason.cs

You could have equivalent for explicit restores.

Reply:
Disadvantange is obviously that you wouldn't get the telemetry for pc projects, but that's not probably not a big deal.

Reply:
@nkolev92 I changed the telemetry, you think this is better? I only know about two ways to do explicit restore, you know if there are more?